snapshot/restore/restore.go

```diff
@@ -203,19 +229,27 @@ func (c *copier) copyEntryInternal(ctx context.Context, e fs.Entry, targetPath s
 	case fs.File:
 		log(ctx).Debugf("file: '%v'", targetPath)
 
-		c.stats.RestoredFileCount.Add(1)
-		c.stats.RestoredTotalFileSize.Add(e.Size())
+		bytesExpected := e.Size()
+		bytesWritten := int64(0)
+		progressCallback := func(chunkSize int64) {
+			bytesWritten += chunkSize
+			c.stats.RestoredTotalFileSize.Add(chunkSize)
```

Comment:
nit; this callback is hit each time we have read `chunkSize` from the repository, not necessarily `RestoredTotalFileSize`. So, `RestoredTotalFileSize ` might become equal to `EnqueuedTotalFileSize` and we could still be writing. Maybe we can introduce a different stat here like `ProcessingTotalFileSize` or `PrefetchedTotalFileSize` ?

Reply:
Yes, that's correct. This progress calculation way treats bytes read as bytes written.